shapeshift/web
# feat: the big short

...ening of Cosmos SDK hacky LoC in profit of opportunities abstraction

## Description



## Notice




## Pull Request Type


## Issue (if applicable)

N/A

## Risk



Low to medium, every single thing Cosmos staking should be tested:
- accounts, account and asset pages
- dashboard balances
- DeFi section cards, rows and balances
- DeFi modals with multi-account, full flow including broadcasting a deposit, withdraw and claim Tx
- staking opportunities in account and asset pages, both with no active opportunity (default validators) and active opportunities
- Earn opportunities without a Cosmos-supporting wallet
- User data refetch on Tx confirmed

Tested all of these and apart from the expected amounts discrepancies (expected because of the precision difference), there should be no noticeable difference vs. prod as a user

Also note that FOXy is rugged on localhost currently, deduct the FOXy balances when testing against prod

See screenshots below

## Testing



- See risks
- Ensure your persisted store is cleared

### Engineering



- ☝🏽 

### Operations



- ☝🏽 

## Screenshots (if applicable)

### Accounts page

#### Prod

<img width="1040" alt="image" src="https://user-images.githubusercontent.com/17035424/217358611-fb1281ec-c82c-445e-962d-a7a42b760755.png">

#### This diff

<img width="1036" alt="image" src="https://user-images.githubusercontent.com/17035424/217358648-ae46a2c3-59ce-4d43-9995-30b7fb8b89a3.png">

### Asset page

#### Prod

<img width="885" alt="image" src="https://user-images.githubusercontent.com/17035424/217358800-50c138f9-0c6b-4782-b3b3-bf38755d134f.png">

#### This diff

<img width="881" alt="image" src="https://user-images.githubusercontent.com/17035424/217358762-f549b039-fd3e-4094-9113-c2fc9454aa0e.png">

### Account 1 page

#### Prod

<img width="889" alt="image" src="https://user-images.githubusercontent.com/17035424/217359362-671e3c21-3e57-4c24-b94e-47c19afc277b.png">

#### This diff

<img width="878" alt="image" src="https://user-images.githubusercontent.com/17035424/217359405-f314421f-8ab3-4224-95cc-869849c80af6.png">

### DeFi overview

#### Prod

<img width="1289" alt="image" src="https://user-images.githubusercontent.com/17035424/217370070-eeae6718-9dfb-45e6-9959-9d994cb048e1.png">

#### This diff - note FOXy is rugged locally so the actual diff vs. prod is really less than 1$ (again something something precision)

<img width="1280" alt="image" src="https://user-images.githubusercontent.com/17035424/217370151-aefd1d98-1e20-4717-a2fa-f3e906483bf2.png">

### Dashboard

#### Prod

<img width="884" alt="image" src="https://user-images.githubusercontent.com/17035424/217368793-ae19f9c4-d7d3-402b-aec3-0f551b4f41a0.png">

#### This diff - note FOXy is rugged locally so the actual diff vs. prod is really less than 1$ (again something something precision)

<img width="873" alt="image" src="https://user-images.githubusercontent.com/17035424/217369677-2448d36c-0bfe-452b-8215-9796e77167ea.png">
